File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalDailyScheduleBridge.java
package com.krystianwsul.checkme.domainmodel.local;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.krystianwsul.checkme.domainmodel.DailyScheduleBridge;
import com.krystianwsul.checkme.persistencemodel.DailyScheduleRecord;
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord;
import com.krystianwsul.checkme.utils.CustomTimeKey;

class LocalDailyScheduleBridge extends LocalScheduleBridge implements DailyScheduleBridge {
    @NonNull
    private final DailyScheduleRecord mDailyScheduleRecord;

    LocalDailyScheduleBridge(@NonNull ScheduleRecord scheduleRecord, @NonNull DailyScheduleRecord dailyScheduleRecord) {
        super(scheduleRecord);

        mDailyScheduleRecord = dailyScheduleRecord;
    }

    @Nullable
    @Override
    public CustomTimeKey getCustomTimeKey() {
        if (mDailyScheduleRecord.getCustomTimeId() != null)
            return new CustomTimeKey(mDailyScheduleRecord.getCustomTimeId());
        else
            return null;
    }

    @Nullable
    @Override
    public Integer getHour() {
        return mDailyScheduleRecord.getHour();
    }

    @Nullable
    @Override
    public Integer getMinute() {
        return mDailyScheduleRecord.getMinute();
    }

    @Override
    public void delete() {
        mScheduleRecord.delete();
        mDailyScheduleRecord.delete();
    }
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalDailyScheduleBridge.kt
package com.krystianwsul.checkme.domainmodel.local

import com.krystianwsul.checkme.domainmodel.WeeklyScheduleBridge
import com.krystianwsul.checkme.persistencemodel.DailyScheduleRecord
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord
import com.krystianwsul.checkme.utils.CustomTimeKey
import com.krystianwsul.checkme.utils.time.DayOfWeek

class LocalDailyScheduleBridge(scheduleRecord: ScheduleRecord, private val mDailyScheduleRecord: DailyScheduleRecord) : LocalScheduleBridge(scheduleRecord), WeeklyScheduleBridge {

    override val customTimeKey get() = mDailyScheduleRecord.customTimeId?.let { CustomTimeKey.LocalCustomTimeKey(it) }

    override val hour get() = mDailyScheduleRecord.hour

    override val minute get() = mDailyScheduleRecord.minute

    override fun delete() {
        scheduleRecord.delete()
        mDailyScheduleRecord.delete()
    }

    override val daysOfWeek = DayOfWeek.values()
            .map { it.ordinal }
            .toSet()
}
